python_modules/dagster-graphql/dagster_graphql/schema/solids.py

```diff
@@ -431,7 +431,10 @@ class Meta:
 
     def __init__(self, represented_pipeline: RepresentedPipeline, solid_def_name: str):
         check.inst_param(represented_pipeline, "represented_pipeline", RepresentedPipeline)
-        self._solid_def_snap = represented_pipeline.get_node_def_snap(solid_def_name)
+        _solid_def_snap = represented_pipeline.get_node_def_snap(solid_def_name)
+        if not isinstance(_solid_def_snap, SolidDefSnap):
```

Comment:
Should we use `check.inst` here?

Reply:
`check.inst` cannot statically narrow (type checker does not get a return type different than the input type)